src/components/organisms/UiTabs/UiTabs.vue

```diff
@@ -75,32 +80,61 @@ const itemsToRender = computed<TabsProps['items']>(() => (props.items.map((item,
     name: name || `tabs-item-${key}`,
   };
 })));
-const tabs = ref<HTMLDivElement | null>(null);
-const activeTabHTMLElement = ref<HTMLElement | null>(null);
-const offsetX = computed(() => {
-  if (activeTabHTMLElement.value === null) return 0;
-  const firstTab = tabs.value?.children[0].children[0];
-  const activeTabRect = activeTabHTMLElement.value.getBoundingClientRect();
-  const tabsRect = tabs.value?.getBoundingClientRect();
-  if (!firstTab) return 0;
-  if (tabsRect && Math.ceil((firstTab.getBoundingClientRect().right / tabsRect.right) * 100) > 50) {
-    return ((activeTabRect.x + activeTabRect.width) - firstTab.getBoundingClientRect().right);
-  }
-  return activeTabRect.x - firstTab.getBoundingClientRect().x;
+
+const tabs = ref<OrNull<HTMLDivElement>>(null);
+const activeTabEl = ref<OrNull<HTMLElement>>(null);
+const offsetX = ref(0);
+const offsetY = ref(0);
+
+const containerRect = reactive({
+  w: 0,
+  y: 0,
+  h: 0,
 });
+
+const calculateOffset = () => {
+  const firstTab = tabs.value?.children[0].children[0];
+  if (!activeTabEl.value || !firstTab) return;
+  const activeRect = activeTabEl.value.getBoundingClientRect();
+  const firstRect = firstTab.getBoundingClientRect();
+
+  offsetX.value = activeRect.x - firstRect.x;
+  offsetY.value = activeRect.y - containerRect.y;
+};
+
+watchEffect(calculateOffset);
+
 const scale = computed(() => {
-  if (activeTabHTMLElement.value === null) return 1;
+  if (activeTabEl.value === null) return 1;
   const firstTabRect = tabs.value?.children[0].children[0].getBoundingClientRect();
   if (!firstTabRect) return 1;
-  const activeTabRect = activeTabHTMLElement.value.getBoundingClientRect();
+  const activeTabRect = activeTabEl.value.getBoundingClientRect();
   return activeTabRect.width / firstTabRect.width;
 });
+
+const containerObserver = new ResizeObserver(([ { target } ]) => {
+  const rect = target.getBoundingClientRect();
```

Comment:
nothing important but :D why you wouldn't use restructuring? 
```
const { width, height, y } = target.getBoundingClientRect();
containerRect.w = width;
containerRect.h = height;
containerRect.y = y;
```
;)

Reply:
I did it on purpose because the linter forces me to split width, height and y into multiple lines

Reply:
yes, fine. We keen object and arrays one element per line. IMO is useful for CR when you add/remove/change something. In our approach, you can find these changes faster with everything in one line. You should find wich element are changed XD in line